Bottom: b313d95e071f70ad95534ac801587b70e42cf32a
Top:    e72e8c85c8f0a2420106f5ba658bb494de9a4429
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-26 12:57:58 +0000

Attempt to debug islandness after inserting hole bug


---

diff --git a/src/action.c b/src/action.c
index f760ecf..5c9fc48 100644
--- a/src/action.c
+++ b/src/action.c
@@ -1508,12 +1508,19 @@ NotifyMode (void)
 		   */
 		  SaveUndoSerialNumber ();
 		  Flags = ((PolygonType *)Crosshair.AttachedObject.Ptr2)->Flags;
-		  PolyToPoursOnLayer (PCB->Data, (LayerType *)Crosshair.AttachedObject.Ptr1,
-				      result, Flags);
+#warning FIXME: Removing a pour should force a re-check of islanding!
+                  /* For now, do this before PolyToPours() which re-inits the
+                   * clipping for the pour(s) affected (?). Ideally RemoveObject
+                   * should fix any island response its self. Anyway.. removing
+                   * the old thing first saves us doing twice the work by
+                   * recomputing islands when it gets deleted.
+                   */
 		  RemoveObject (POUR_TYPE,
 				Crosshair.AttachedObject.Ptr1,
 				Crosshair.AttachedObject.Ptr2,
 				Crosshair.AttachedObject.Ptr3);
+		  PolyToPoursOnLayer (PCB->Data, (LayerType *)Crosshair.AttachedObject.Ptr1,
+				      result, Flags);
 		  RestoreUndoSerialNumber ();
 		  IncrementUndoSerialNumber ();
 		  Draw ();
